rust/connlib/libs/client/src/file_logger.rs

```diff
@@ -0,0 +1,52 @@
+//! Connlib File Logger
+//!
+//! This module implements a file-based logger for connlib using tracing-appender.
+//!
+//! The log files are rotated hourly.
+//!
+//! Since these will be leaving the user's device, these logs should contain *only*
+//! the necessary debugging information, and **not** any sensitive information,
+//! including but not limited to:
+//! - WiFi SSIDs
+//! - Location information
+//! - Device names
+//! - Device serials
+//! - MAC addresses
+
+use std::path::PathBuf;
+use tracing::{level_filters::LevelFilter, Subscriber};
+use tracing_subscriber::Layer;
+
+const LOG_FILE_BASE_NAME: &str = "connlib.log";
+
+/// Create a new file logger layer.
+pub fn layer<T>(
+    log_dir: PathBuf,
+) -> (
+    Box<dyn Layer<T> + Send + Sync + 'static>,
+    tracing_appender::non_blocking::WorkerGuard,
+)
+where
+    T: Subscriber + for<'a> tracing_subscriber::registry::LookupSpan<'a>,
+{
+    tracing::info!("Saving log files to: {}", log_dir.display());
```

Comment:
Will this ever be printed? Since you're probably using this when creating the subscriber I'm not sure.

Reply:
It depends on whether the client has another logging mechanism or not set up. On Android, since `android_logger` is configured before tracing, I think it does. But I'm cool with removing it since the behavior wouldn't be consistent.